### PR TITLE
fix(statusChatInput): do not replace text with emojis on links

### LINKED ISSUE #8478: symbols in the pasted url link are replaced by emoji
when inserting URL in the chat, some symbols might be replaced by emoji so the link is invalid, and its not possible to open it

### LINKED ISSUE #8750: Chat Input: Http detection is misbehaving and fails to detect emoji
### Description
Repro steps:

Type http :D
Send message

Actual: emoji not converted

https://user-images.githubusercontent.com/47811206/207103885-ea0dc740-f65c-4141-b1b3-ff0fe902eb6c.mov

### PR DESCRIPTION
**Pull request by alexandraB99**
Closes #8478
Closes #8750

### What does the PR do
status chat input prevents adding emoji when links are pasted 

### Affected areas
status chat input

https://user-images.githubusercontent.com/31625338/219080580-327eac9c-5fd4-4f20-a877-0a5cb4265701.mov

### CODE REVIEW

File: ui/imports/shared/status/StatusChatInput.qml [hunk 1]
```diff
@@ -1207,17 +1206,15 @@ Rectangle {
 
                 RowLayout {
                     Layout.fillWidth: true
-                    Layout.fillHeight: true
+                    Layout.minimumHeight: messageInputField.contentHeight
```
Comment (caybro) [inline ui/imports/shared/status/StatusChatInput.qml:1209]: This works much better :)

```suggestion
                    Layout.minimumHeight: messageInputField.contentHeight + messageInputField.topPadding + messageInputField.bottomPadding
```

Comment (caybro) [reply ui/imports/shared/status/StatusChatInput.qml:1209]: Without it, the chat input height is still funky with multi line text

File: ui/imports/shared/status/StatusChatInput.qml [hunk 2]
```diff
@@ -1285,13 +1282,14 @@ Rectangle {
...
                                         const emojis = StatusQUtils.Emoji.emojiJSON.emoji_json.filter(function (emoji) {
-                                            if (emoji.aliases_ascii.includes(getText((cursorPosition - 2), cursorPosition)) ||
-                                                    emoji.aliases_ascii.includes(getText((cursorPosition - 3), cursorPosition))) {
-                                                var has2Chars = emoji.aliases_ascii.includes(getText((cursorPosition - 2), cursorPosition));
-                                                replaceWithEmoji("", getText(cursorPosition - (has2Chars ? 2 : 3), cursorPosition), emoji.unicode);
+                                            if (emoji.aliases_ascii.includes(getText((symbolIdx-1), (symbolIdx+1))) ||
+                                                emoji.aliases_ascii.includes(getText((symbolIdx-2), (symbolIdx+1)))) {
+                                                var has2Chars = emoji.aliases_ascii.includes(getText((symbolIdx-4), (symbolIdx-2)));
+                                                replaceWithEmoji("", getText((symbolIdx - (has2Chars ? 3 : 2)), symbolIdx), emoji.unicode);
```
Comment (caybro) [inline ui/imports/shared/status/StatusChatInput.qml:1292]: I still can't get the :beers: emoji; if I start typing `:b`, this gets immediately replaced with an unrelated emoji:

image

**Review by caybro — CHANGES_REQUESTED** [review]

**Comment by alexjba** [comment]
Looks like it's also fixing #8750

File: ui/imports/shared/status/StatusChatInput.qml (hunk 2 above)
Comment (alexjba) [reply ui/imports/shared/status/StatusChatInput.qml:1292]: I think it's an older issue. I can't send beers on 0.10

Comment (alexjba) [reply ui/imports/shared/status/StatusChatInput.qml:1292]: It's hard to maintain both emoji shortName detection and ascii aliases. It's inevitable to have collisions while typing. Might be a design question on how it should work while typing.
I see Discord is using only shortName detection in the chat input, but the discord chat history will try to detect both ascii alias and short name. Sounds reasonable to me.

**Review by alexjba — APPROVED** [review]
LGTM! Other than the `minimumHeight` update suggested by @caybro
Might be worth creating another task to clarify the emoji detection while typing with design team. The Discord approach sounds better to me:
1. Detect emoji short name in chat input
2. Detect both short name and ascii alias in chat history

WDYT?

File: ui/imports/shared/status/StatusChatInput.qml (hunk 2 above)
Comment (caybro) [reply ui/imports/shared/status/StatusChatInput.qml:1292]: I think both are doable; we'd just have to do the inline replacement when having typed `:b ` (ie. only after a space); I also noticed that Discord starts presenting the popup after having typed at least 2 characters from the emoji alias (ie. only after `:be`), not on the first char after colon (`:`)

**Comment by caybro** [comment]
> LGTM! Other than the `minimumHeight` update suggested by @caybro Might be worth creating another task to clarify the emoji detection while typing with design team. The Discord approach sounds better to me:
> 
> 1. Detect emoji short name in chat input
> 2. Detect both short name and ascii alias in chat history
> 
> WDYT?

I'll repeat it here too: 
I think both are doable; we'd just have to do the inline replacement when having typed :b  (ie. only after a space); I also noticed that Discord starts presenting the popup after having typed at least 2 characters from the emoji alias (ie. only after :be), not on the first char after colon (:)

File: ui/imports/shared/status/StatusChatInput.qml (hunk 2 above)
Comment (alexjba) [reply ui/imports/shared/status/StatusChatInput.qml:1292]: > > LGTM! Other than the `minimumHeight` update suggested by @caybro Might be worth creating another task to clarify the emoji detection while typing with design team. The Discord approach sounds better to me:
> > 
> > 1. Detect emoji short name in chat input
> > 2. Detect both short name and ascii alias in chat history
> > 
> > WDYT?
> 
> I'll repeat it here too: I think both are doable; we'd just have to do the inline replacement when having typed :b (ie. only after a space); I also noticed that Discord starts presenting the popup after having typed at least 2 characters from the emoji alias (ie. only after :be), not on the first char after colon (:)

Good point 👍. Sounds like a good rule

**Comment by alexandraB99** [comment]
@caybro I have added the paddings in the minimum height. As about the rest, I was also thinking that this is something to be clarified with design, this PR is only focusing on preventing the conversion to emoji of combination characters included in a link. The proposed solution sounds good, I will open a new issue regarding the behavior overall

**Comment by alexandraB99** [comment]
updated issue https://github.com/status-im/status-desktop/issues/8465 adding the latest proposals

**Review by caybro — APPROVED** [review]
LGTM

The overall inplace emoji editting will be handled separately

**Pull request merged**